server/chat/backend/agent/tools/artifact_tool.py

```
@@ -0,0 +1,177 @@
+
+
+class ReadArtifactArgs(BaseModel):
+    title: str = Field(description="The exact title of the artifact to read")
+
+
+class WriteArtifactArgs(BaseModel):
+    title: str = Field(description="The exact title of the artifact to create or update")
+    content: str = Field(description="The full markdown content of the document")
+
+
+def list_artifacts(user_id: str | None = None, **kwargs) -> str:
```

Ruff (0.15.15)

[warning] Missing type annotation for `**kwargs`

(ANN003)